internal/query/result_set_range_test.go

```diff
@@ -0,0 +1,830 @@
+//go:build go1.22 && goexperiment.rangefunc
+
+package query
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ydb-platform/ydb-go-genproto/protos/Ydb"
+	"github.com/ydb-platform/ydb-go-genproto/protos/Ydb_Query"
+	"go.uber.org/mock/gomock"
+	grpcCodes "google.golang.org/grpc/codes"
+	grpcStatus "google.golang.org/grpc/status"
+
+	"github.com/ydb-platform/ydb-go-sdk/v3/internal/xerrors"
+	"github.com/ydb-platform/ydb-go-sdk/v3/internal/xtest"
+)
+
+func TestResultSetRange(t *testing.T) {
```

Comment:
Need additional test for `break` behavior.
I expect to close the result set and I can iterate for next result set in the result.